aspnetcore/security/authentication/certauth.md

```diff
@@ -548,3 +548,28 @@ namespace AspNetCoreCertificateAuthApi
     }
 }
 ```
+
+<a name="occ"></a>
+
+## Optional client certificates
+
+This section provides information for apps that must protect a subset of the app with a certificate. For example, a Razor Page or controller in the app might require client certificates. This presents challenges as client certificates:
+  
+* Are a TLS feature, not an HTTP feature.
+* Are negotiated per-connection and must be be negotiated at the start of the connection before any HTTP data is available. At the start of the connection, only the Server Name Indication (SNI)&dagger; is known. The client and server certificates are negotiated prior to the first request on a connection and requests generally won't be able to renegotiate. Renegotiation is prohibited in HTTP/2.
+
+ASP.NET Core 5 preview 4 and later supports optional client certificates. For more information, see the [Optional certificates sample](https://github.com/dotnet/aspnetcore/tree/master/src/Security/Authentication/Certificate/samples/Certificate.Optional.Sample).
+
+The following approach supports optional client certificates:
+
+* Set up binding for the domain and subdomain:
+  * For example, set up bindings on `contoso.com` and `myClient.contoso.com`. The `contoso.com` host doesn't require a client certificate but `myClient.contoso.com` does.
+  * For more information, see `[TODO - provide links]()`
```

Comment:
See for Kestrel https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel?view=aspnetcore-3.1#listenoptionsusehttps and HttpsConnectionAdapterOptions.ClientCertificateMode 

And for IIS https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/?view=aspnetcore-3.1#create-the-iis-site and https://docs.microsoft.com/en-us/iis/manage/configuring-security/how-to-set-up-ssl-on-iis#configure-ssl-settings-2

For Http.Sys see https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/httpsys?view=aspnetcore-3.1#configure-windows-server and https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/cc725882(v=ws.10)?redirectedfrom=MSDN#add-sslcert clientcertnegotiation